### DISCUSSION TITLE
Confused on some GitHub commands

### DISCUSSION THREAD

**Discussion by DatDerpyDude81** [community/community#33383] [Other Feature Feedback, Questions, & Ideas] [answered]
Can someone tell me the difference between `git push` and `git commit`? Apparently `git commit` only commits to local repository but then why do I see commits on Github?

**Comment by zarifpour** [comment, marked as answer]
```
The “git commit” command is used to tell Git to save your changes to the local repository and you have to specifically tell Git which changes you wish to include in a commit before using the “git commit” command. You can use the “git push” command to push the commits on the local repository to a remote repository.
```

Practically speaking, when you commit to your local repository those changes will only be available locally and will only be synced with GitHub, or whichever remote you are using, when you do a `git push`.

So the order of operations here would be to:

1. Make small manageable commits with descriptive messages
2. Push changes to remote when ready to sync your local changes with your remote repository

It's possible you already pushed some commits before or submitted changes directly on GitHub, which is why you are seeing a commit history in your repository.

**Reply by DatDerpyDude81** [reply]
Thanks!
